controller/api/authentication.py

```diff
@@ -9,3 +10,15 @@ def authenticate(self, request):
         Authenticate the request for anyone!
         """
         return AnonymousUser(), None
+
+
+class AnonymousOrAuthenticatedAuthentication(authentication.BaseAuthentication):
+
+    def authenticate(self, request):
+        """
+        Authenticate the request for anyone or if a valid token is provided, a user.
+        """
+        try:
+            return TokenAuthentication.authenticate(TokenAuthentication(), request)
```

Comment:
Yeah I guess it's fine since we're hooking into just Token auth for registration right now but we will hit a scalability problem in the future as we add more pluggable auth strategies.